### PR TITLE
fix for #3068 : Node logs are flooded with SigarExceptions on some windows machines

### PR DESCRIPTION
**Pull request by fviale**
- As these exceptions are printed by the sigar libary directly, we have to redirect error stream before calling sigar's getAttribute
- refactored the code for clarity

### CODE REVIEW

File: rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RRDSigarDataStore.java [hunk 1]
```diff
@@ -213,4 +199,86 @@ void sample(RrdDb dataBase, long timeInMs) throws IOException {
...
+        private ByteArrayOutputStream storedErrors;
+
+        private PrintStream redirectedStream;
+
+        private PrintStream console;
+
+        public PrintStream redirectStdErr() throws UnsupportedEncodingException {
+            storedErrors = new ByteArrayOutputStream();
+            redirectedStream = new PrintStream(storedErrors, true, StandardCharsets.UTF_8.name());
+            console = System.err;
```
Comment (sonarqube-activeeon) [inline rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RRDSigarDataStore.java]: MAJOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RRDSigarDataStore.java [hunk 2]
```diff
@@ -170,41 +174,23 @@ void sample(RrdDb dataBase, long timeInMs) throws IOException {
...
+            readAttribute(timeInMs, sample, dataSource, fullName, attrName, objectName);
+
+            String errorContent = stdErrRedirectionHelper.readAndReset();
+
+            if (errorContent.length() > 0) {
+                logErrorOnce(errorContent,
+                             "Error while reading attribute " + attrName + " for object " + objectName + ": " +
+                                           System.lineSeparator() + errorContent);
             }
         }
```
Comment (gheon) [inline rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RRDSigarDataStore.java]: Don't you need to stop the redirection here?

Comment (fviale) [reply rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RRDSigarDataStore.java]: the method readAndReset stops the redirection

File: rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RRDSigarDataStore.java (hunk 1 above)
Comment (fviale) [reply rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RRDSigarDataStore.java]: no I won't do that, dude ...

**Pull request merged**